var/spack/repos/spack_repo/builtin/packages/neovim/package.py

```diff
@@ -91,78 +93,81 @@ class Neovim(CMakePackage):
         description="use lua rather than luajit as lua language provider",
     )
 
-    depends_on("c", type="build")  # generated
-
-    # depend on virtual, lua-luajit-openresty preferred
-    depends_on("lua-lang")
-    depends_on("luajit", when="~no_luajit")
-    depends_on("lua-lang@5.1", when="+no_luajit")
-
-    # dependencies to allow regular lua to work
-    depends_on("lua-ffi", when="^[virtuals=lua-lang] lua", type=("link", "run"))
-    depends_on("lua-bitlib", when="^[virtuals=lua-lang] lua", type=("link", "run"))
-
-    # base dependencies
+    depends_on("c", type="build")
     depends_on("cmake@3.0:", type="build")
     depends_on("pkgconfig", type="build")
+
     depends_on("gettext")
     depends_on("gperf", type="link")
+    depends_on("iconv", type="link")
     depends_on("jemalloc", type="link", when="platform=linux")
+    depends_on("libtermkey", type="link")
+    depends_on("libluv", type="link")
+    depends_on("libuv", type="link")
+    depends_on("libvterm", type="link")
+    depends_on("lua-lang")
```

Comment:
I see you removed the conditionality on these dependencies. IIRC this ends up adding superfluous dependencies on some (and newer) versions, since some of them have been dropped.

I documented a bit in #50123 what has been dropped/updated (far from being complete and perfect).

Moreover, since this PR is a "style" one, I would go for keeping everything as is from a semantic point of view.